src/App.tsx

```diff
@@ -8,37 +8,108 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getUsers } from './utils/getUsers';
+import { getPosts } from './utils/getPosts';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [sidebarOpen, setSidebarOpen] = useState(false);
+
+  const loadUser = async () => {
+    try {
+      setHasError(false);
+
+      const userFromServer = await getUsers();
+
+      setUsers(userFromServer);
+    } catch {
+      setIsLoading(false);
```

Comment:
move this state change into finally block, because loading should be ended in any case

Reply:
fix everywhere